ipfs/ipfs.go

```diff
@@ -0,0 +1,72 @@
+package ipfs
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"mime/multipart"
+	"net/http"
+)
+
+const (
+	addAPI = "api/v0/add"
+)
+
+// IPFS is the struct that contains all informations to access an IPFS node
+type IPFS struct {
+	client   *http.Client
+	endpoint string
+}
+
+// Response return the response of a file uploaded
+type Response struct {
+	Name string
+	Hash string
+	Size string
+}
+
+// New creates a new IPFS client method
+func New() *IPFS {
+	return &IPFS{
+		client:   &http.Client{},
+		endpoint: "https://ipfs.infura.io:5001/",
+	}
+}
+
+// Add data from a reader to the IPFS node
+func (ipfs *IPFS) Add(name string, reader io.Reader) (*Response, error) {
+	var b bytes.Buffer
```

Comment:
this will memory a lot. instead we can just use [http.Post()](https://golang.org/pkg/net/http/#Client.Post). since it accepts an _io.Reader_, we can directly pass this _reader_ var to it. for the _content-type_, it'll be the same for every upload, so we can use a constant value